### DISCUSSION TITLE
File Naming Convention and Examples Concerning Plan Name

### DISCUSSION THREAD

**Discussion by ghost** [CMSgov/price-transparency-guide#35] [Q&A] [answered]
The file naming convention is

"The following is the required naming standard for each file: <YYYY-MM-DD>_<payer or issuer name>_<plan name>_<file type name>.json"

The examples provider are
For example, the following would be the required naming for CMS:
2020-01-05_cms_medicare_in-network-rates.json
2020-01-05_cms_medicare_allowed-amounts.json
2020-01-05_cms_medicare_prescription-drugs.json

The <plan name> in the file name should not be the same value used in the 3 files schema definition of plan_name which is
"The plan name and name of plan sponsor and/or insurance company." which would result in a file named as
2020-01-05_cms_medicarecms_in-network-rates.json

Just verifying that plan name is being defined differently between the file naming convention and the schema field for plan name.

**Comment by tdfow** [comment, marked as answer]
Since the schema has the HIOS ID/EIN  (plan_id) associated with the 'plan_name', wouldn't the plan_name need to associate with the file name?  Otherwise, how does the schema support multiple plan_id information in one file, as the reporting information, plan_name and plan_id seems to identify/associate to the file name.    I was confused by the file naming standard examples provided too.

**Reply by ghost** [reply]
Honestly based on the schema, json examples and file naming convention/examples there is not a single file say for a in-network-rates file but instead there will be multiple. Because of the schema top level layout of plan/HIOS that has supporting specific array of In-Network objects, a different plan for a insurer will need to have its own file. They only way you could get around it to a single file would be to have the that plan specific fields in the schema placed in a Plan object array that would then encapsulate the plan's associated In-Network objects that are associated with it.

**Reply by ghost** [reply]
Based on the example JSON for in-network-rates file it would be ,
{
  "reporting_entity_name": "ABC Health Insurance",
  "reporting_entity_type": "health insurance issuer",
  "plan_name": "medicare",
  "plan_id_type": "HIOS",
  "plan_id": "12345XX9876543",
  "plan_market_type": "individual",
  "last_updated_on": "2020-08-27",
  "in_network": []
}
with a file name of 2020-08-27_abchealthinsurance_medicare_in-network-rates.json

then the insurance of course supports other plans they provide like a plan called "Healthy Plan"
{
  "reporting_entity_name": "ABC Health Insurance",
  "reporting_entity_type": "health insurance issuer",
  "plan_name": "healthy plan",
  "plan_id_type": "HIOS",
  "plan_id": "1775XX9876543",
  "plan_market_type": "group",
  "last_updated_on": "2020-08-27",
  "in_network": []
}
with a file name of 2020-08-27_abchealthinsurance_healthyplan_in-network-rates.json

**Reply by shaselton-usds** [reply]
This is correct in the understanding of individual files are produced based on the number of plans offered. There isn't the expectation that a single file would contain all plans that require reporting -- one plan per file. The file naming is basically correct. A clarification on turning spaces found in the `plan_name` into dashes will be added to the main README.md. So, in this case, your two file names would be: `2020-08-27_abc-health-insurance_medicare_in-network-rates.json` and `2020-08-27_abc-health-insurance_healthyplan_in-network-rates.json`

I think I see the confusion around the definition of the `plan_name` element required in the schema vs the README.md documentation where there are basic universal instructions.

**Comment by tdfow** [comment]
That has been my interpretation---that the Reporting Entity name and plan_name should align with the file naming standard provided, but hoping that can be clarified and the naming standard reference be updated to identify expectations.
